### PR TITLE
tests: add mariaDB to version upgrading CI

### PR DESCRIPTION
**Pull request by lance6716**
### What problem does this PR solve? 
part of https://github.com/pingcap/dm/issues/1280

### What is changed and how it works?
as title

### Check List 

Tests 

 - No code

Related changes

 - Need to cherry-pick to the release branch

### CODE REVIEW

**Comment by lance6716** [comment]
/hold

```
Starting component `dmctl`: /root/.tiup/components/dmctl/v2.0.0/dmctl/dmctl --master-addr=master1:8261 operate-source create /go/src/github.com/pingcap/dm/tests/tiup/conf/source2.yaml
{
    "result": false,
    "msg": "[code=10001:class=database:scope=not-set:level=high], Message: database driver error, RawCause: sql: no rows in result set, Workaround: Please check the database connection and the database config in configuration file.",
    "sources": [
    ]
}
```

case "ha"

```
task test has no source or not exist, please check the task name and status
```

**Comment by lance6716** [comment]
the failed test case should be fixed by https://github.com/pingcap/dm/pull/1616

File: tests/tiup/upgrade-from-v2.sh [hunk 1]
```diff
@@ -30,6 +30,7 @@ function migrate_in_previous_v2() {
 	# v2.0.0 doesn't support relay log
 	if [[ "$PRE_VER" == "v2.0.0" ]]; then
 		sed -i "s/enable-relay: true/enable-relay: false/g" $CUR/conf/source2.yaml
+		sed -i "s/enable-gtid: true/enable-gtid: false/g" $CUR/conf/source2.yaml
```
Comment (lichunzhu) [inline tests/tiup/upgrade-from-v2.sh:33]: Why do we set `enable-gtid` to false here? Could you please leave a comment?

**Review by lichunzhu — COMMENTED** [review]
/lgtm

**Pull request merged**